### PR TITLE
Update gradle

### PR DESCRIPTION
**Pull request by domdomegg**
Updated gradle build tools to 2.3.0, updated android support libs to v25.2, updated targetSdkVersion to 25. Removed unused ndk libs, code that required it was removed several months ago.

More info in #381 

Also was having a look at some of the other libs and had some questions:
1. On every gradle build I get: 
    ```
    WARNING: Dependency org.apache.httpcomponents:httpclient:4.2.2 is ignored for release as it may be conflicting with the internal version provided by Android.
   ```
   and I'm not sure when it comes from. It seems unrelated to the `org.apache.http.legacy` lib on line 29, as removing that line makes no change. Doesn't seem to be doing anything bad, just might be something that might be adding bulk to the app for no reason.

2. Do we still need to be using the old version of GSON? See #410

### CODE REVIEW

**Comment by domdomegg** [comment]
Nope - clearly NDK was required by something - will have a look in a sec

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
```
Comment (maskaravivek) [inline build.gradle]: Gradle constants should be used for these version numbers.

**Comment by misaochan** [comment]
Would you like to refactor it to use constants @domdomegg ? Granted we haven't been doing that this whole time, haha, but from what I'm reading it's probably good practice to do so.

**Comment by domdomegg** [comment]
This must have somehow slipped past me - didn't see the review. Yes I think it would be good to use constants, I'll try to update it when I get home.

**Comment by domdomegg** [comment]
Should be good now?

**Comment by misaochan** [comment]
Looks good to me! Would you like to merge, @veyndan ? :)

**Comment by domdomegg** [comment]
Dunno - thought I had resolved all the conflicts to merge from Github anyways. Hey, it worked though :P

**Comment by veyndan** [comment]
There was a conflict in the latest commit i.e. [https://github.com/commons-app/apps-android-commons/pull/411/commits/51d3bd85b5b823eeb56a5e8547e2caaffc3f30a2](https://github.com/commons-app/apps-android-commons/pull/411/commits/51d3bd85b5b823eeb56a5e8547e2caaffc3f30a2).

**Comment by veyndan** [comment]
Why was this reopened?

**Comment by misaochan** [comment]
Sorry about that. I was wondering why it said 'closed' instead of 'merged', but the commits are indeed there. Might be better to merge via the web interface next time?

**Comment by veyndan** [comment]
Yeah sorry, I noted that in a previous comment

**Pull request closed without merging**